File: java-sandbox/java-maven-tests/src/main/java/com/example/sandbox/BankAccount.java
package com.example.sandbox;

/** Small stateful class to demonstrate lifecycle hooks and exception checks. */
public class BankAccount {

    private double balance;

    public BankAccount(double balance) {
        this.balance = balance;
    }

    public double getBalance() {
        return balance;
    }

    public void deposit(double amount) {
        if (amount <= 0) {
            throw new IllegalArgumentException("amount must be positive");
        }
        balance += amount;
    }

    public void withdraw(double amount) {
        if (amount > balance) {
            throw new IllegalStateException("insufficient funds");
        }
        balance -= amount;
    }
}

File: csharp-sandbox/cs-nunit-tests/src/BankAccount.cs
namespace Sandbox;

/// <summary>Small stateful class to demonstrate lifecycle hooks and exceptions.</summary>
public class BankAccount
{
    public BankAccount(decimal balance) => Balance = balance;

    public decimal Balance { get; private set; }

    public void Deposit(decimal amount)
    {
        if (amount <= 0)
        {
            throw new ArgumentException("amount must be positive", nameof(amount));
        }

        Balance += amount;
    }

    public void Withdraw(decimal amount)
    {
        if (amount > Balance)
        {
            throw new InvalidOperationException("insufficient funds");
        }

        Balance -= amount;
    }
}
